internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -238,6 +252,14 @@ func (u *Upgrader) Upgrade(ctx context.Context, version string, sourceURI string
 	}
 
 	archivePath, err := u.downloadArtifact(ctx, parsedVersion, sourceURI, det, skipVerifyOverride, skipDefaultPgp, pgpBytes...)
+
+	// If the artifactPath is not empty, then the artifact was downloaded.
+	// There may still be an error in the download process, so we need to add
+	// downloads path to the cleanup slice.
+	if archivePath != "" {
+		cleanupPaths = append(cleanupPaths, paths.Downloads())
```

Comment:
?? why are we checking archive path when appending download path, also when making os.RemoveAll on cleanup path where `paths.Download` is present, we need to make sure that during download missing download path is not problematic